core/src/test/java/org/elasticsearch/index/fielddata/IndexFieldDataServiceTests.java

```diff
@@ -238,4 +188,67 @@ public void onRemoval(ShardId shardId, MappedFieldType.Names fieldNames, FieldDa
         }
     }
 
+
+    public void testRequireDocValuesOnLongs() {
+        ThreadPool threadPool = new ThreadPool("random_threadpool_name");
```

Comment:
These seem copy-and-paste-ish. Can you make a method that does the assertion and just call it for all the types? Or do it in a for loop or something?